src/test/java/com/commercetools/project/sync/shoppinglist/ShoppingListSyncerTest.java

```diff
@@ -50,20 +57,35 @@ void of_ShouldCreateShoppingListSyncerInstance() {
   @Test
   void transform_ShouldReplaceShoppingListReferenceIdsWithKeys() {
     // preparation
+    final SphereClient sourceClient = mock(SphereClient.class);
     final ShoppingListSyncer shoppingListSyncer =
-        ShoppingListSyncer.of(
-            mock(SphereClient.class), mock(SphereClient.class), mock(Clock.class));
+        ShoppingListSyncer.of(sourceClient, mock(SphereClient.class), mock(Clock.class));
     final List<ShoppingList> shoppingList =
         Collections.singletonList(readObjectFromResource("shopping-list.json", ShoppingList.class));
 
+    mockSourceClient(sourceClient);
+
     // test
     final CompletionStage<List<ShoppingListDraft>> draftsFromPageStage =
         shoppingListSyncer.transform(shoppingList);
 
     // assertion
     assertThat(draftsFromPageStage)
         .isCompletedWithValue(
-            ShoppingListReferenceResolutionUtils.mapToShoppingListDrafts(shoppingList));
+            ShoppingListTransformUtils.toShoppingListDrafts(
+                    sourceClient, referenceIdToKeyCache, shoppingList)
+                .join());
+  }
+
+  private void mockSourceClient(SphereClient sourceClient) {
```

Comment:
those utilities are duplicated between tests, would be better to not share the same snippet everywhere in tests and use common util? You might pass the key/result as a parameter.

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/268/commits/a8b3f521c7417bbe5d7ce011c3f5327f8553226e